src/utils/daos_hdlr.c

```diff
@@ -2519,6 +2524,8 @@ fs_copy(struct file_dfs *src_file_dfs,
 					fprintf(stderr, "updating destination "
 						"permissions failed on %s "
 						"(%d)\n", next_dpath, rc);
+					D_FREE(next_path);
```

Comment:
rather than duplicating these frees, one could just do something like 

if (rc != 0) {
    D_FREE(...);
    D_FREE(...); 
}

At the bottom.  As long as they are NULL initialized, it is safe to call D_FREE on a NULL pointer.

Reply:
Sure, I moved them to the bottom.